syntax = "proto3";

import "svg.proto";

package mr_proto;

/* Map Render settings */
message MapRenderSettings {
    double width = 1;               // ширина изображения в пикселях. Вещественное число в диапазоне от 0 до 100000.
    double height = 2;              // высота изображения в пикселях. Вещественное число в диапазоне от 0 до 100000.
    double padding = 3;             // отступ краёв карты от границ SVG-документа. Вещественное число не меньше 0 и меньше min(width, height)/2.
    double line_width = 4;          // толщина линий, которыми рисуются автобусные маршруты. Вещественное число в диапазоне от 0 до 100000.
    double stop_radius = 5;         // радиус окружностей, которыми обозначаются остановки. Вещественное число в диапазоне от 0 до 100000.
    uint32 bus_label_font_size = 6; // размер текста, которым написаны названия автобусных маршрутов. Целое число в диапазоне от 0 до 100000.
    svg_proto.Point bus_label_offset = 7;     // смещение надписи с названием маршрута относительно координат конечной остановки на карте. 
                                    // Массив из двух элементов типа double. Задаёт значения свойств dx и dy SVG-элемента <text>.
                                    // Элементы массива — числа в диапазоне от –100000 до 100000.
    uint32 stop_label_font_size = 8;// размер текста, которым отображаются названия остановок. Целое число в диапазоне от 0 до 100000.
    svg_proto.Point stop_label_offset = 9;    // смещение названия остановки относительно её координат на карте. Массив из двух элементов типа double.
                                    // Задаёт значения свойств dx и dy SVG-элемента <text>. Числа в диапазоне от –100000 до 100000.
    svg_proto.Color underlayer_color = 10;    // цвет подложки под названиями остановок и маршрутов. Формат хранения цвета будет ниже.
    double underlayer_width = 11;   // толщина подложки под названиями остановок и маршрутов. Задаёт значение атрибута stroke-width элемента <text>. Вещественное число в диапазоне от 0 до 100000.
    repeated svg_proto.Color color_palette = 12;  // цветовая палитра. Непустой массив.
}
